python/events/evaluation_tools/events/event_detection/decomposition.py

```diff
@@ -0,0 +1,168 @@
+"""
```

Comment:
While addressing the below two comments, it would not hurt to add type hints in the method declarations. For example:

```python
def detrend_streamflow(series: pd.Series, halflife: Union[float, str, pd.Timedelta], window: Union[int, pd.offsets, pd.Index], reverse: bool =False) -> pd.Series:
```